### ISSUE TITLE
SEB crash repeatedly on automatic startup in Win10

### ISSUE THREAD

**Issue by abowman-unige** [SafeExamBrowser/seb-win-refactoring#580]
We are currently experiencing random crash when SEB is automatically starting on startup in our virtual machine. Not on every case but when SEB crash sometimes it can't be started again, it repeatedly crash the same way. Fyi: SEB is started via GPO after user log on.

 - OS: Windows 10
 - SEB-Version 3.4.0

Here's some logs where on two occasions we managed to gather information:
[seb_crash_startup_case1.zip](https://github.com/SafeExamBrowser/seb-win-refactoring/files/10922398/seb_crash_startup_case1.zip)
[seb_crash_startup_case2.zip](https://github.com/SafeExamBrowser/seb-win-refactoring/files/10922402/seb_crash_startup_case2.zip)

**Comment by dbuechel** [comment]
The two crashes are most likely completely unrelated. In one case, SEB was not able to open a communication channel from one of its components to another. In the other case, SEB seems to have crashed during runtime where there appears to have been running one or multiple prohibited processes:

```
2023-02-08 11:49:45.379 [14] - DEBUG: [ApplicationMonitor] Window has changed from 'Taskbar' (131944) to 'Taskbar' (131298).
2023-02-08 11:49:45.397 [16] - WARNING: [ApplicationMonitor] Window 'Taskbar' (131298) belongs to not whitelisted process 'dwm.exe'!
2023-02-08 11:49:45.398 [16] - WARNING: [ApplicationMonitor] Failed to hide window 'Taskbar' (131298)!
2023-02-08 11:49:45.967 [14] - DEBUG: [ApplicationMonitor] Window has changed from 'Taskbar' (131298) to 'Microsoft Windows' (393246).
2023-02-08 11:49:45.989 [24] - WARNING: [ApplicationMonitor] Window 'Microsoft Windows' (393246) belongs to not whitelisted process 'dwm.exe'!
2023-02-08 11:49:45.989 [24] - WARNING: [ApplicationMonitor] Failed to hide window 'Microsoft Windows' (393246)!
2023-02-08 11:49:46.002 [16] - INFO: [ApplicationMonitor] Sent close message to window 'Taskbar' (131298).
2023-02-08 11:49:46.191 [24] - INFO: [ApplicationMonitor] Sent close message to window 'Microsoft Windows' (393246).
2023-02-08 11:49:46.198 [14] - DEBUG: [ApplicationMonitor] Window has changed from 'Microsoft Windows' (393246) to 'Taskbar (Ne répond pas)' (131298).
2023-02-08 11:49:46.216 [21] - WARNING: [ApplicationMonitor] Window 'Taskbar (Ne répond pas)' (131298) belongs to not whitelisted process 'dwm.exe'!
2023-02-08 11:49:46.217 [21] - WARNING: [ApplicationMonitor] Failed to hide window 'Taskbar (Ne répond pas)' (131298)!
```

To investigate the latter case, we'd need the error entry or entries from the application log of the Windows event viewer utility. The former issue is most likely caused due to a not-yet-completely-ready system or framework, as you said that you're trying to start SEB automatically after startup. There once was another issue related to this (see #32), what I know from our internal setup is that we have a timeout of about 30 seconds in place after system startup and before actually starting SEB, so you may want to try doing that as well.

**Comment by dbuechel** [comment]
@abowman-unige Were you able to obtain the system event logs in the meantime?

**Comment by dbuechel** [comment]
We just fixed another, possibly related issue (#576). You may want to test the latest beta build in order to see whether the changes also fixed your problem: https://sebdev-let.ethz.ch/api/buildjobs/du6h6x4a9duy0bd6/artifacts/SEB_3.5.0.543_SetupBundle.exe.

**Comment by abowman-unige** [comment]
Thank you, we will try the beta asap. In the meantime we are optimizing our virtual machines master for a faster boot on user login and did some test that seems to work until know. 

I don't know if it helps seb at any point but we check this option in the seb service policy for the seb local configuration and it seems to wait on the service to start up ?

![image](https://user-images.githubusercontent.com/127325893/228457219-af2a8309-86bc-42fb-ad31-34f3e9a1decf.png)

**Comment by dbuechel** [comment]
Yes, as that service policy requires the SEB Service to be running in order to start a new session, see https://safeexambrowser.org/windows/win_usermanual_en.html#SecurityPane.
